Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import querystring from 'node:querystring';
+import memoize from 'fast-memoize';
+import { create as createJwt } from 'njwt';
+import { Logger } from '@edfi/meadowlark-core';
+import { admin1, client1, client2, client3, client4 } from '../security/HardcodedCredential';
+import type { Jwt } from '../security/Jwt';
+import type { AuthorizationResponse } from './AuthorizationResponse';
+import type { AuthorizationRequest } from './AuthorizationRequest';
+import type { RequestTokenBody } from '../model/RequestTokenBody';
+import type { BodyValidation } from '../validation/BodyValidation';
+import { validateRequestTokenBody } from '../validation/ValidateRequestTokenBody';
+import type { GetAuthorizationClientResult } from '../message/GetAuthorizationClientResult';
+import { ensurePluginsLoaded, getAuthorizationStore } from '../plugin/AuthorizationPluginLoader';
+import { hashClientSecretHexString } from '../security/HashClientSecret';
+
+const moduleName = 'RequestToken';
+
+type ParsedRequestTokenBody =
+  | { isValid: true; requestTokenBody: RequestTokenBody }
+  | { isValid: false; failureMessage: string };
+
+function signingKey(): Buffer {
+  const signingKeyEncoded = process.env.SIGNING_KEY;
+  if (signingKeyEncoded == null) {
+    throw new Error('Must have a base-64 encoded signing key. Try creating a new one with `yarn createKey`');
+  }
+  return Buffer.from(signingKeyEncoded, 'base64');
+}
+const cachedSigningKey = memoize(signingKey);
+
+/*
+ * Creates a standard Meadowlark Jwt.
+ */
+function createToken(clientId: string, vendor: string, roles: string[]): Jwt {
+  const claims = { iss: 'ed-fi-meadowlark', aud: 'meadowlark', roles, client_id: clientId };
```

Comment:
I wonder if these should be config values?

Reply:
I'm going to create a separate ticket that will cover:

1. moving these to config values
2. documenting the oauth server setup